.github/actions/detect-workflow-js/src/main.ts

```diff
@@ -0,0 +1,118 @@
+import * as github from "@actions/github";
+import * as core from "@actions/core";
+import type { Endpoints } from "@octokit/types";
+import path from "path";
+
+type ApiWorkflowRun =
+  Endpoints["GET /repos/{owner}/{repo}/actions/runs/{run_id}"]["response"]["data"];
+
+type githubClaimsType = {
+  aud?: string;
+  job_workflow_ref?: string;
+};
+
+function decodeToken(federatedToken: string): githubClaimsType {
+  const tokenPayload = federatedToken.split(".")[1];
+  const bufferObj = Buffer.from(tokenPayload, "base64");
+  const decoded = JSON.parse(bufferObj.toString("utf8"));
+  return decoded;
+}
+
+async function run(): Promise<void> {
+  /* Test locally. Requires a GitHub token:
+        $ env INPUT_TOKEN="$(gh auth token)" \
+        GITHUB_RUN_ID="4289844197" \
+        GITHUB_REPOSITORY="konstruktoid/ansible-role-dns" \
+        nodejs ./dist/index.js
+    */
+  const token = core.getInput("token");
+  const octokit = github.getOctokit(token);
+
+  const repoName = process.env.GITHUB_REPOSITORY;
+  if (!repoName) {
+    core.setFailed("No repository detected.");
+    return;
+  }
+
+  // Set outputs.
+  let repository = "";
+  let ref = "";
+  let workflow = "";
+
+  try {
```

Comment:
nit: I would perhaps split these out into functions rather than have a large if/else-like try/catch block.

We could more easily unit test each function that way as well.